src/utils/selectedCellUtils.ts

```diff
@@ -64,19 +63,23 @@ export function getNextSelectedCellPosition<R, SR>({
         if (colIdx > posIdx) break;
         const colSpan = getColSpan(column, lastFrozenColumnIndex, { type: 'ROW', row });
         if (colSpan && posIdx > colIdx && posIdx < colSpan + colIdx) {
-          position.idx = colIdx + (moveRight ? colSpan : 0);
-          break;
+          return {
+            ...position,
```

Comment:
It's just 1 property, so we can set it manually and avoid the iterating over the object by spreading it.
```suggestion
            rowIdx: position.rowIdx,
```